widget/table.go

```diff
@@ -154,6 +180,43 @@ func (t *Table) templateSize() fyne.Size {
 	return fyne.Size{}
 }
 
+func (t *Table) visibleColumnWidths(colWidth, cols int) (visible map[int]int, offX, minCol, maxCol int) {
+	maxCol = cols
+	colOffset := 0
+	isVisible := false
+	visible = make(map[int]int)
+
+	if t.scroll.size.Width <= 0 {
+		return
+	}
+
+	for i := 0; i < cols; i++ {
+		width := colWidth
+		if w, ok := t.columnWidths[i]; ok {
+			width = w
+		}
+
+		if colOffset <= t.offset.X-width-tableDividerThickness {
+			// before scroll
+		} else if colOffset <= t.offset.X {
+			minCol = i
+			offX = colOffset
+			isVisible = true
```

Comment:
Should this get reset to false if the column is after the scroll viewport, or are you just relying on the break on 209?

Reply:
Relying on break - we are starting from 0 so it should not be possible that we pass the window without realising